webview-ui/test-generation/src/App.tsx

```diff
@@ -115,26 +121,59 @@ function App() {
     status,
     steps,
     tunnel,
+    loadedAppInfo,
   } = state;
 
+  const loadedAppNames: string[] = loadedAppInfo.map((map) => map.name);
+  const appNameToPlatform = new Map<string, string>();
+  loadedAppInfo.forEach((item) => appNameToPlatform.set(item.name, item.kind));
+  let seenHistoryAppName = appName == ''; // if appName is empty, we don't need to add appName later
+  const optionElements: JSX.Element[] = [];
+  for (let i = 0; i < loadedAppNames.length; i++) {
+    optionElements.push(<VSCodeOption>{loadedAppNames[i]}</VSCodeOption>);
+    if (loadedAppNames[i] == appName) {
+      seenHistoryAppName = true;
+    }
+  }
+  if (!seenHistoryAppName) {
+    optionElements.push(
+      <VSCodeOption className="app-not-loaded">{appName}</VSCodeOption>,
+    );
+  }
+
   return (
     <main>
       <section className="inputs">
         <h2>What do you want to test?</h2>
-        <VSCodeTextField
-          value={appName}
-          placeholder="From Sauce Labs App Storage, e.g. test.apk"
-          onInput={(e) => {
-            if (e.target && 'value' in e.target) {
-              dispatch({
-                type: 'setAppName',
-                value: e.target.value as string,
-              });
+        <section className="with-label">
+          <label>App Name</label>
+          <VSCodeDropdown
+            onInput={(e) => {
+              if (e.target && 'value' in e.target) {
+                const appName: string = e.target.value as string;
+                dispatch({
+                  type: 'setAppName',
+                  value: {
+                    appName: appName as string,
+                    platformName: appNameToPlatform.get(appName) as
+                      | 'ios'
+                      | 'android',
+                  },
+                });
+              }
+            }}
+            value={appName}
+            className={
+              optionElements.some(
+                (option) => option.props.className === 'app-not-loaded',
+              )
+                ? 'app-not-loaded'
+                : ''
             }
```

Comment:
You can use the `:has()` pseudo-class instead of this.

css like this:
```css
.app-list:has(.app-not-loaded) {
  color: gray;
}
```
Then just add the `.app-list` class:

```suggestion
            className="app-list"
```

Reply:
This doesn't quite have the functionality I want, instead of only greying out the "bad" app, when I'm on that history instance, it greys out every app I selected, and the "bad" app is not greyed out in the drop down list

Reply:
😕 the two implementations are behaving identically for me...